compiler/rustc_error_codes/src/error_codes/E0733.md

```diff
@@ -13,7 +13,7 @@ async fn foo(n: usize) {
 To perform async recursion, the `async fn` needs to be desugared such that the
```

Comment:
It feels preferable to change the suggestion to suggest `Box::pin(recur()).await` as the first option (noting that it requires a newer version), this is a fairly significant improvement

Reply:
while I would still like to change it to use `Box::pin` as the first suggestion, this is not blocking :shrug: